monitor/services/src/main/java/org/datacleaner/monitor/server/components/ComponentHandler.java

```diff
@@ -0,0 +1,251 @@
+        lifeCycleHelper.validate(descriptor, component);
+        lifeCycleHelper.initialize(descriptor, component);
+    }
+
+    public List<Object[]> runComponent(JsonNode data) {
+        List<InputRow> inputRows;
+        DataSetHeader header = new SimpleDataSetHeader(table.getColumns());
+        inputRows = new ArrayList<>();
+        int id = 0;
+        for (JsonNode row : data) {
+            DefaultRow inputRow = new DefaultRow(header, toRowValues((ArrayNode) row));
+            inputRows.add(new MetaModelInputRow(id, inputRow));
+            id++;
+        }
+
+        // TODO: run with the engine thread-pool executor
+        if(component instanceof Transformer) {
+            List results = new ArrayList();
+            for (InputRow inputRow : inputRows) {
+                results.add(((Transformer)component).transform(inputRow));
+            }
+            return results;
+        } else if(component instanceof Analyzer) {
+            for (InputRow inputRow : inputRows) {
+                ((Analyzer)component).run(inputRow, 1);
+            }
+            return null;
+        } else {
+            throw new IllegalArgumentException("Unknown component type " + component.getClass());
+        }
+    }
+
+    public AnalyzerResult closeComponent() {
+        LifeCycleHelper lifeCycleHelper = new LifeCycleHelper(dcConfiguration, null, false);
+        lifeCycleHelper.close(descriptor, component, true);
+        if(component instanceof HasAnalyzerResult) {
+            return ((HasAnalyzerResult)component).getResult();
+        } else {
+            return null;
+        }
+    }
+
+    private InputColumn getOrCreateInputColumn(String columnName) {
+        MutableColumn column = columns.get(columnName);
+        InputColumn inputColumn;
+        if(column == null) {
+            columns.put(columnName, column = new MutableColumn(columnName, ColumnType.VARCHAR, table, columns.size(), true));
+        }
+        inputColumn = inputColumns.get(columnName);
+        if(inputColumn == null) {
+            inputColumns.put(columnName, inputColumn = new MetaModelInputColumn(column));
+        }
+        return inputColumn;
+    }
+
+    private Object[] toRowValues(ArrayNode row) {
+        ArrayList<Object> values = new ArrayList<>();
+        int i = 0;
+        for(JsonNode value: row) {
+            if(i >= table.getColumnsCount()) {
+                LOGGER.debug("Data contain more columns than specified. Will be ignored.");
+                break;
+            }
+            Column col = table.getColumn(i);
+            values.add(convertTableValue(col.getType().getJavaEquivalentClass(), value));
+        }
+        return values.toArray(new Object[values.size()]);
+    }
+
+    private Object convertPropertyValue(ConfiguredPropertyDescriptor propDesc, JsonNode userPropValue) {
+        // TODO: other properties type. Find some generic way to convert JsonNode to the specific property type
```

Comment:
Generic way would be to use StringConverter from DC-engine-core. This is the same class that is used when reading analysis.xml files and I think using the same syntax is much preferable .... Even also for InputColumns (which don't get into this method currently).